### PR TITLE
Add flash messages to CMS pages

### PR DESCRIPTION
**Pull request by murny**
fixes this issue: https://github.com/ualbertalib/discovery/issues/1443

- Removed 'blacklight' layout in favour of 'application' layout (nothing was using this...so lets get rid of one layout in favour of rails defaults)

Improvements:
- Why are we using 4 layouts for the CMS? Should be able to get away with maybe one? maybe two (cms and french cms?)? Read docs here on handling locale: https://github.com/comfy/comfortable-mexican-sofa/wiki/Docs:-Translations
- lots of duplication in these layouts (home/french home are not using the shared header partial etc)
- we got a 'wrapper' div...which closing tag is in the footer partial? And we have a `push` div...which isnt doing much at all...
- french layout has two google analytics scripts?

Above improvements captured in this ticket: https://github.com/ualbertalib/discovery/issues/1447

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,3 +1,57 @@
...
+    <!--[if IEMobile]>
+      <meta http-equiv="cleartype" content="on">
+    <![endif]-->
+
+    <title>University of Alberta Libraries</title>
+    <%= opensearch_description_tag application_name, opensearch_catalog_url(:format => 'xml') %>
+    <%# favicon_link_tag asset_path('favicon.ico') %>
+    <%= favicon_link_tag 'favicon.ico' %>
+    <%= stylesheet_link_tag    "application", "https://fonts.googleapis.com/css?family=Open+Sans:400,700,400italic|Abel", "sticky-footer", "ualib-discovery", media: "all" %>
+    <%= javascript_include_tag "application", "stupidtable", "ualib-discovery" %>
```
Comment (pgwillia) [inline app/views/layouts/application.html.erb]: was `<%= javascript_include_tag "application", "ualib-home" %>` Why this change?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+
+  </head>
+  <body class="<%= render_body_class %>" data-no-turbolink="true">
+    <div class="wrapper">
+  <%= render :partial => 'shared/header_navbar' %>
+
+  <%= render partial: 'shared/ajax_modal' %>
+
+  <div id="main-container" class="container">
+    <%= render partial: '/flash_msg', layout: 'shared/flash_messages' %>
```
Comment (pgwillia) [inline app/views/layouts/application.html.erb]: Which file does 'shared/flash_messages' refer to?

Comment (pgwillia) [reply app/views/layouts/application.html.erb]: Ah!  It's in blacklight https://github.com/projectblacklight/blacklight/blob/v5.15.0/app/views/shared/_flash_messages.html.erb

**Comment by murny** [comment]
Yeah good catches. Will add a changelog. 

Test be nice...but theirs some setup there (like getting someway to provision the CMS in the test environment etc? Heck this be nice to setup for development too!)

This will be tough and you bring up a good point. When library-cms gets pulled away, we will have to circle back and figure out a better way to handle this. Because they will then be two different apps they will not be able to share the flash messages in this way. So maybe a better approach would be instead of redirecting them to the `cms homepage` we can redirect them to the `catalog index`? Not sure... Needs some thinking for this transition (which is a bigger discussion about how the CMS and Discovery going to coexist as other things will need to change as well, etc)

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (murny) [reply app/views/layouts/application.html.erb]: This diff was me deleting `blacklight.html.erb` layout and moving everything over into `application.html.erb` as application.html.erb was never used and is rails default.

So this is just whatever blacklight.html.erb had.

`<%= javascript_include_tag "application", "ualib-home" %>` will include the javascript for both the application and ualib-home manifests. Which ideally should only be one javascript file... `application` not two. 

Some more refactoring could take place here.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (murny) [reply app/views/layouts/application.html.erb]: Yep! I was in the progress of just using our own flash message partial... As exactly...where does this come from? Why we need a partial with a layout etc? but Blacklight references this flash partial for some of their javascript code. So figured better not change it in case it breaks a few things.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (pgwillia) [reply app/views/layouts/application.html.erb]: I must have done something funny when I was looking at a diff of this file and blacklight.html.erb. No concerns here

**Comment by murny** [comment]
Changelog entry added :+1:
